src/plugins/intel_cpu/src/nodes/topk.cpp

```diff
@@ -2006,8 +2019,10 @@ void TopK::prepareParams() {
         // [case 1]: if 2 * (top_k + 1) + 2 <= count_xmm, thus top_k is small enough that the vector registers are sufficient
         //           to keep all necessary data for sorting, no need to load and store frequently, use inplace bubble sort;
         //           (horizotal sorting cases not included)
-        // [case 2]: only when topk is imposed on innermost dimsension of planar(ncsp/nspc) layout, should heap sort be used;
-        // [case 3]: by default, use bitonic sort when alg_cost_bitonic < alg_cost_bubble, otherwise use bubble sort.
+        // [case 2]: if stable sorting is required, bubble sort will be applied currently, because among the implemented sorting
```

Comment:
As I understand, bubble sort is used for unstable algorithm as well and in general there is no warranty that unstable behavior is aligned with reference or frameworks. Looks like it was a luck that we were not faced with repeating values (or second output was not used in such cases).

Reply:
Thanks Nikolay for the commet! You are right about that not all the implemented bubble sorts are stable. However, in fact, as the dimensions (21, 21, 21, 21) are quite large in tests, generic generator also make repeating input values. The reason why all tests pass is that, https://github.com/openvinotoolkit/openvino/pull/16522/commits/e6d4a234699a2b206abb1dedb92821ee3d4ea855 (_Invoke topk_bubble_horiz as unstable sort_) had been used to ensure that only stable bubble sort will be applied if the stable flag is set. Anyway, I've updated the description to avoid petential confusion. Further discussions are always welcomed! Thanks!

Reply:
Applied. Thanks Nikolay!